src/library_wasmfs_opfs.js

```diff
@@ -343,9 +343,13 @@ mergeInto(LibraryManager.library, {
   },
 
   _wasmfs_opfs_set_size_access__deps: ['$wasmfsOPFSAccessHandles'],
-  _wasmfs_opfs_set_size_access: async function(ctx, accessID, size) {
+  _wasmfs_opfs_set_size_access: async function(ctx, accessID, size, errPtr) {
```

Comment:
In a PR I am writing I used a return value for the error, using the standard 0 for success and error otherwise. Is there a reason not to do it that way?

Reply:
Since this function is async, it actually returns a promise and you would have to be able to await that promise to get the return value you intended.

Reply:
An async function can return a non-promise value too (it's ok to `await 42`). So I think we could do it that way too?

Reply:
Well, maybe it's simpler this way, I don't feel strongly.

Reply:
Yes, but only if it returns before it does its first `await`, since `await` requires suspending the computation and returning to the event loop.

Reply:
I think it works even after an await. It's ok to mix Promise results with non-Promise results. E.g. this works:
```js
function timeout(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

async function foo(x) {
  if (x == 1) {
    console.log('a');
    await timeout(1000);
    console.log('b');
    await timeout(1000);
    console.log('c');
  }
  return 10;
}

async function main() {
  console.log(await foo(1));
  console.log(await foo(2));
}

main();
```
That has a non-Promise return after an await, and the internal awaits are also conditional.

Reply:
But that wouldn't work if an `await` was actually executed before the `return 10`, would it?

Reply:
An `await` (well, 2 of them) is executed if `x == 1` there, which is the first case that `main` does.

Reply:
Oh right. But `main` does an `await foo(1)`. I'm pretty sure that `foo(1)` is returning a promise...

Reply:
Ah, ok, I guess that internally then an async function returning `42` turns that `42` into a resolved Promise of `42`...

Reply:
And IIUC, awaiting still requires a return to the microtask queue even the value is immediately available, so there's really no way around returning via a pointer here.